runtime/lua/vim/_editor.lua

```diff
@@ -1176,9 +1176,85 @@ function vim._init_default_autocmds()
   })
 end
 
+--- Parse a string of hex chars as a color.
+---
+--- The string can contain 1 to 4 hex characters. The returned value is a
+--- fractional value between 0 and 1 representing the intensity of the color in
+--- the range.
+---
+--- For instance, if only a single hex char "a" is used, then this function
+--- returns 0.625 (10 / 16), while a value of "aa" would return 0.664 (170 /
+--- 256).
+---
+--- @param c string Color as a string of hex chars
+--- @return number Intensity of the color
+local function hexcolor(c)
+  local len = #c
+  if not c:match('^0x') then
+    c = string.format('0x%s', c)
+  end
+
+  local max = tonumber(string.format('0x%s', string.rep('f', len)))
+  return tonumber(c) / max
+end
+
+--- Guess value of 'bg' based on terminal color.
+---
+--- We write Operating System Command (OSC) 11 to the terminal to request the
+--- terminal's background color. We then wait for a response. If the response
+--- matches `rgba:RRRR/GGGG/BBBB/AAAA` where R, G, B, and A are hex digits, then
+--- compute the luminance[1] of the RGB color and classify it as light/dark
+--- accordingly. Note that the color components may have anywhere from one to
+--- four hex digits, and require scaling accordingly as values out of 4, 8, 12,
+--- or 16 bits. Also note the A(lpha) component is optional, and is parsed but
+--- ignored in the calculations.
+--- 
+--- [1] https://en.wikipedia.org/wiki/Luma_%28video%29
+local function guess_background_color()
+  if vim.api.nvim_get_option_info2('bg', {}).was_set then
+    -- Don't do anything if 'bg' is already set
+    return
+  end
+
+  vim.api.nvim_create_autocmd('TermResponse', {
```

Comment:
TODO: Add a timer to delete this autocommand if no response is received after some timeout